internal/clients/account.go

```diff
@@ -62,12 +73,63 @@ func (account *ResourceManagerAccount) GetSubscriptionId() string {
 	}
 
 	if account.subscriptionId == nil {
-		log.Printf("[DEBUG] No default subscription ID found")
+		log.Printf("[DEBUG] No subscription ID found")
 		return ""
 	}
+
 	return *account.subscriptionId
 }
 
+func (account *ResourceManagerAccount) GetObjectId() string {
+	account.mutex.Lock()
+	defer account.mutex.Unlock()
+
+	if account.objectId != nil {
+		return *account.objectId
+	}
+
+	tok, err := account.client.Option.Cred.GetToken(account.client.StopContext, policy.TokenRequestOptions{
+		TenantID: account.client.Option.TenantId,
+		Scopes:   []string{account.client.Option.CloudCfg.Services[cloud.ResourceManager].Endpoint + "/.default"}})
+	if err != nil {
+		log.Printf("[DEBUG] Error getting requesting token from credentials: %s", err)
+	}
+
+	cl, err := parseTokenClaims(tok.Token)
```

Comment:
Do we need a nil-check on the `tok`? If above GetToken returns error, the `tok` could be nil here?

Reply:
`tok` is of type [AccessToken](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore@v1.13.0/internal/exported#AccessToken). This is a value type not a pointer.

The type declaration is:

```go
type AccessToken struct {
	Token     [string](https://pkg.go.dev/builtin#string)
	ExpiresOn [time](https://pkg.go.dev/time).[Time](https://pkg.go.dev/time#Time)
}
```

I refactored using a comparison of `tok.Token == ""` as this will be the default value if the token request fails.

WDYT?

[b7ae1bb](https://github.com/Azure/terraform-provider-azapi/pull/657/commits/b7ae1bb25967bce4ddf8a785ddf0d1255bc89e91)